policy_definitions/tagging/policy_definition_qby_require_tags.json

```
@@ -0,0 +1,41 @@
+{
+    "name": "QBY-Require-Tag",
+    "type": "Microsoft.Authorization/policyDefinitions",
+    "apiVersion": "2021-06-01",
+    "properties": {
+        "policyType": "Custom",
+        "mode": "All",
+        "displayName": "Require defined tags on resources.",
+        "description": "Require resources to have conceptual required tags and not empty.",
+        "parameters": {
+            "tagName": {
+                "type": "String",
+                "metadata": {
+                    "displayName": "Tag Name",
+                    "description": "Name of the tag required."
+                }
+            }
+        },
+        "policyRule": {
+            "if": {
+                "allOf": [
+                    {
+                        "anyOf": [
+                            {
+                                "field": "[concat('tags[', parameters('tagName'), ']')]",
+                                "exists": false
+                            },
+                            {
+                                "field": "[concat('tags[', parameters('tagName'), ']')]",
+                                "equals": ""
+                            }
+                        ]
+                    }
+                ]
+            },
```

```suggestion
            "if": {
                      "anyOf": [
                          {
                              "field": "[concat('tags[', parameters('tagName'), ']')]",
                              "exists": false
                          },
                          {
                              "field": "[concat('tags[', parameters('tagName'), ']')]",
                              "equals": ""
                          }
                      ]
            },
```

the all of is useless if it only contains one element

---

Resolved as suggestion.